test/blackbox/BlackboxTests.cpp

```diff
@@ -5269,6 +5269,23 @@ BLACKBOXTEST(BlackBox, AsyncVolatileKeepAllPubReliableSubNonReliableHelloWorld)
     reader.block_for_at_least(2);
 }
 
+// Regression test of Refs #3376, github ros2/rmw_fastrtps #226
+BLACKBOXTEST(BlackBox, ReqRepVolatileHelloworldRequesterCheckWriteParams)
+{
+    ReqRepAsReliableHelloWorldRequester requester;
+    ReqRepAsReliableHelloWorldReplier replier;
+
+    requester.durability_kind(eprosima::fastrtps::VOLATILE_DURABILITY_QOS).init();
```

Comment:
I think the replier should also be set to volatile for this test to work

Reply:
The replier is not needed in this test. I removed it.

Reply:
LGTM